lib/src/util/coap_util.dart

```diff
@@ -7,88 +7,8 @@
 
 part of coap;
 
-/// Cancellable asynchronous sleep support class
-class CoapCancellableAsyncSleep {
-  CoapCancellableAsyncSleep(this._timeout);
-
-  final Duration _timeout;
-
-  /// Timeout
-  Duration get timeout => _timeout;
-
-  /// The completer
-  final Completer<void> _completer = Completer<void>();
-
-  /// The timer
-  late Timer _timer;
-
-  /// Timer running flag
-  bool _running = false;
-
-  /// Running
-  bool get isRunning => _running;
-
-  /// Start the timer
-  Future<void> sleep() {
-    if (!_running) {
-      _timer = Timer(_timeout, _timerCallback);
-      _running = true;
-    }
-    return _completer.future;
-  }
-
-  /// Cancel the timer
-  void cancel() {
-    if (_running) {
-      _timer.cancel();
-      _running = false;
-      _completer.complete();
-    }
-  }
-
-  /// The timer callback
-  void _timerCallback() {
-    _running = false;
-    _completer.complete();
-  }
-}
-
 /// Utility methods
 class CoapUtil {
-  /// Insertion sort, to make the options list stably ordered.
-  static void insertionSort<T>(List<T> list, Comparator<T> comparison) {
-    collection.insertionSort(list, compare: comparison);
-  }
-
-  /// Checks if all items in both of the two enumerables are equal.
-  static bool areSequenceEqualTo<T>(Iterable<T>? first, Iterable<T>? second,
-      [collection.Equality<T>? equality]) {
-    final ie = collection.IterableEquality<T>(equality!);
-    return ie.equals(first, second);
-  }
-
-  /// Finds the first matched item.
-  /// Returns the item found, or null if none is matched.
-  static T? firstOrDefault<T>(Iterable<T> source, bool Function(T) condition) =>
-      source.firstWhereOrNull(condition);
-
-  /// Checks if matched item exists.
-  /// Returns true if exists any matched item, otherwise false.
-  static bool contains<T>(Iterable<T> source, bool Function(T) condition) =>
-      source.takeWhile(condition).isNotEmpty;
-
-  /// Stringify an iterable.
-  static String iterableToString<T>(Iterable<T> source) {
-    final sb = StringBuffer();
-    for (final item in source) {
-      sb.write(item.toString());
-      if (item != source.last) {
-        sb.write(',');
-      }
-    }
-    return sb.toString();
-  }
-
   /// Stringify options in a message.
   static String optionsToString(CoapMessage msg) {
     final sb = StringBuffer();
```

Comment:
On second glance, maybe this method could also be moved to the `CoapMessage` class? If I see it correctly, that is the only place where the method is being used.

Reply:
Nice catch, fixed: https://github.com/shamblett/coap/pull/87/commits/7136515b45582fce34bce0c8746722ae2ae23f81